exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcPreparedStatementImpl.kt

```diff
@@ -30,15 +31,19 @@ class JdbcPreparedStatementImpl(val statement: PreparedStatement, val wasGenerat
     }
 
     override fun setNull(index: Int, columnType: IColumnType) {
-        if (columnType is BinaryColumnType || columnType is BlobColumnType) {
+        if (columnType is BinaryColumnType) {
             statement.setNull(index, Types.LONGVARBINARY)
         } else {
             statement.setObject(index, null)
         }
     }
 
     override fun setInputStream(index: Int, inputStream: InputStream) {
-        statement.setBinaryStream(index, inputStream, inputStream.available())
+        if (currentDialect is SQLiteDialect) {
+            statement.setBytes(index, inputStream.readBytes())
+        } else {
+            statement.setBlob(index, inputStream)
+        }
```

Comment:
Please add a parameter to the base function and isolate only the necessary changes:
```kt
override fun setInputStream(index: Int, inputStream: InputStream, setAsBlobObject: Boolean) {
    if (setAsBlobObject) {
        statement.setBlob(index, inputStream)
    } else {
        statement.setBinaryStream(index, inputStream, inputStream.available())
    }
}

// PreparedStatementApi.kt - add parameter to setInputStream() and edit KDocs

/**
  * Sets the statement parameter at the [index] position to the provided [inputStream],
  * either directly as a BLOB if `setAsBlobObject` is `true` or as determined by the driver.
  */
fun setInputStream(index: Int, inputStream: InputStream, setAsBlobObject: Boolean)
```

Reply:
`inputStream.available()` is not usable to get the length of the InputStream, see EXPOSED-86

Reply:
@elektro-wolle Old blob tests are failing on SQLite because the driver requires a version of `setBinaryStream()` that takes a length parameter:
```kt
statement.setBinaryStream(index, inputStream, inputStream.available())
```

Please fix and any issue with `available()` can then be addressed in the dedicated PR for EXPOSED-86.